src/main/java/com/mcsunnyside/coreprotecttnt/Main.java

```diff
@@ -266,213 +298,228 @@ public void onEndCrystalHit(EntityDamageByEntityEvent e) {
-                api.logRemoval("[Fire]" + ignitedBlocks.get(e.getIgnitingBlock().getLocation()), e.getBlock().getLocation(), e.getBlock().getType(), e.getBlock().getBlockData());
-            } else if(getConfig().getBoolean("fire.disable-burn-when-target-not-found")) {
+            if (probablyCache.getIfPresent(e.getIgnitingBlock().getLocation()) != null) {
+                probablyCache.put(e.getBlock().getLocation(), probablyCache.getIfPresent(e.getIgnitingBlock().getLocation()));
+                api.logRemoval("#fire-" + probablyCache.getIfPresent(e.getIgnitingBlock().getLocation()), e.getBlock().getLocation(), e.getBlock().getType(), e.getBlock().getBlockData());
+            } else if (section.getBoolean("disable-unknown", true)) {
                 e.setCancelled(true);
-                Collection<Entity> entityCollections = e.getBlock().getLocation().getWorld().getNearbyEntities(e.getBlock().getLocation(), 15, 15, 15);
-                for (Entity entity : entityCollections) {
-                    if (entity instanceof Player)
-                        entity.sendMessage(
-                                ChatColor.translateAlternateColorCodes('&',
-                                        getConfig().getString("msgs.fire-wont-burn-blocks")
-                                )
-                        );
-                }
+                Util.broadcastNearPlayers(e.getIgnitingBlock().getLocation(), section.getString("alert"));
             }
         }
     }
 
+//    Will make empty records for unknown reason :(
+//    @EventHandler(ignoreCancelled = true, priority = EventPriority.MONITOR)
+//    public void onEntityPickup(InventoryPickupItemEvent e) {
+//        ConfigurationSection section = Util.bakeConfigSection(getConfig(), "entity-pickup");
+//        if (!section.getBoolean("enable", true))
+//            return;
+//        if (!(e.getInventory().getHolder() instanceof Entity))
+//            return;
+//        // Copy from CoreProtect
+//        Item item = e.getItem();
+//        Location location = item.getLocation();
+//        ItemStack itemStack = item.getItemStack();
+//        if (itemStack != null) {
+//            String loggingItemId = "#" + ((Entity) e.getInventory().getHolder()).getName().replace(" ", "_") + "." + location.getBlockX() + "." + location.getBlockY() + "." + location.getBlockZ();
+//            int itemId = CTNTQueue.getItemId(loggingItemId);
+//            List<ItemStack> list = (List) ConfigHandler.itemsPickup.getOrDefault(loggingItemId, new ArrayList());
+//            list.add(itemStack.clone());
+//            ConfigHandler.itemsPickup.put(loggingItemId, list);
+//            int time = (int) (System.currentTimeMillis() / 1000L) + 1;
+//            CTNTQueue.queueItemTransaction("#" + ((Entity) e.getInventory().getHolder()).getName().replace(" ", "_"), location.clone(), time, itemId);
+//        }
+//    }
```

Comment:
Is this really needed for explosion logging?

Reply:
removed